client/api_basic.go

```diff
@@ -142,17 +144,23 @@ func (c *client) WaitForNBlocks(ctx context.Context, n int64) error {
 }
 
 // WaitForTx requests the tx from hash, if not found, waits for next block and
-// tries again. Returns an error if ctx is canceled.
+// tries again. Returns an error if ctx is canceled or the number of waiting blocks exceeds MaxWaitBlockNum blocks.
```

Comment:
What is the rationale behind this requirement?

Reply:
I prefer to use original go function to support timeout.

```
ctxTimeout, cancel := context.WithTimeout(context.Background(), time.Second*3)
defer cancel()
```

Reply:
updated to use timeout for controlling the situation